### PR TITLE
Users class

### PR DESCRIPTION
**Pull request by RayFoje**
add class User and modify class Main

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -2,27 +2,52 @@
 import java.util.*;
 import java.util.stream.IntStream;
 
+class User {
```
Comment (SophiToyber) [inline src/Main.java]: Make user as separate class ( I mean create .java class in inteleji, such as Main.java)

File: src/Main.java [hunk 2]
```diff
@@ -2,27 +2,52 @@
...
         Scanner scanner = new Scanner(System.in);
         System.out.print("Enter the number of users: ");
         int numberUser = scanner.nextInt();
         scanner.nextLine();
-        List<String> userNames = new ArrayList<>();
-        List<Integer> userAges = new ArrayList<>();
+
+        List<User> users = new ArrayList<>();
 
         IntStream.range(0, numberUser).forEach(i -> {
```
Comment (SophiToyber) [inline src/Main.java:13]: Why do u use here IntStream.range()? What you expected form this operation?

File: src/Main.java [hunk 3]
```diff
@@ -32,26 +57,25 @@ public static void main(String[] args) {
                 switch (selectedOption) {
```
Comment (SophiToyber) [inline src/Main.java]: move swith-case into separate function

File: src/Main.java (hunk 2 above)
Comment (RayFoje) [reply src/Main.java:13]: I use it to create a sequence of numbers in a specified range. In this case it is the number of users.  I want to receive the number of users to check the limit

File: src/Main.java (hunk 3 above)
Comment (RayFoje) [reply src/Main.java]: remove)

File: src/Main.java [hunk 4]
```diff
@@ -8,50 +8,30 @@ public static void main(String[] args) {
...
         int numberUser = scanner.nextInt();
         scanner.nextLine();
-        List<String> userNames = new ArrayList<>();
-        List<Integer> userAges = new ArrayList<>();
+
+        List<User> users = new ArrayList<>();
 
         IntStream.range(0, numberUser).forEach(i -> {
-            if (userNames.size() >= numberUser) {
+            if (users.size() >= numberUser) {
```
Comment (SophiToyber) [inline src/Main.java]: You can move it above, before start of the stream range

File: src/Main.java [hunk 5]
```diff
@@ -8,50 +8,30 @@ public static void main(String[] args) {
...
                 System.out.println("You have reached your user limit.");
-                printUserList(userNames, userAges);
+                printUserList(users);
```
Comment (SophiToyber) [inline src/Main.java]: not needed here

File: src/Main.java [hunk 6]
```diff
@@ -8,50 +8,30 @@ public static void main(String[] args) {
...
             System.out.print("Enter age: ");
-            userAges.add(scanner.nextInt());
+            int age = scanner.nextInt();
             scanner.nextLine();
+            users.add(new User(name, age));
+
             System.out.print("Do you want to continue? (y - Yes, n - No): ");
             char selectedExit = scanner.nextLine().charAt(0);
 
             if (selectedExit == 'y') {
```
Comment (SophiToyber) [inline src/Main.java]: No needed her for now, please reove it

File: src/Main.java [hunk 7]
```diff
@@ -60,13 +40,39 @@ public static void main(String[] args) {
         });
 
         System.out.println("You have reached your user limit.");
```
Comment (SophiToyber) [inline src/Main.java]: Alse reove

File: src/Main.java [hunk 8]
```diff
@@ -60,13 +40,39 @@ public static void main(String[] args) {
...
+        printUserList(users);
         System.exit(0);
-            }
+    }
 
-    private static void printUserList(List<String> userNames, List<Integer> userAges) {
-        IntStream.range(0, userNames.size())
-                .forEach(i -> System.out.println("User" + (i + 1) + ": " + userNames.get(i) + ", " + userAges.get(i)));
+    private static void printUserList(List<User> users) {
+        IntStream.range(0, users.size())
```
Comment (SophiToyber) [inline src/Main.java]: You can use user.stram() instead of this  IntStream.range(0, users.size())

File: src/Main.java [hunk 9]
```diff
@@ -60,13 +40,39 @@ public static void main(String[] args) {
...
+                .forEach(i -> System.out.println("User" + (i + 1) + ": " + users.get(i)));
+    }
+
+    private static void processUserOptions(Scanner scanner, List<User> users) {
+        System.out.print("Calculate average age(1), Print only all names(2), Finish and print result user list(3): ");
+        int selectedOption = scanner.nextInt();
+        switch (selectedOption) {
+            case 1:
+                System.out.print("Average age:");
+                double averageAge = users.stream().mapToInt(User::getAge).average().orElse(0);
```
Comment (SophiToyber) [inline src/Main.java:70]: Find more information about this one operation User::getAge  *non-code comment, just for the self-study

File: src/Main.java (hunk 5 above)
Comment (RayFoje) [reply src/Main.java]: remove

File: src/Main.java (hunk 6 above)
Comment (RayFoje) [reply src/Main.java]: remove

File: src/Main.java (hunk 7 above)
Comment (RayFoje) [reply src/Main.java]: remove

File: src/Main.java (hunk 8 above)
Comment (RayFoje) [reply src/Main.java]: complete

**Pull request merged**